File: Java/src/test/java/com/gildedrose/TexttestFixture.java
package com.gildedrose;

public class TexttestFixture {
    public static void main(String[] args) {
        Item[] items = new Item[] {
                new Item("Bananas", 10, 20),
                new Item("Aged Brie", 2, 0),
                new Item("Eggs", 5, 7),
                new Item("Eggs", 12, 5),
                new Item("Canned Beans", 0, 80),
                new Item("Canned Beans", -1, 80),

                // This Baked good does not work properly yet!
                new Item("Baked Sourdough Bread", 3, 6) };

        GildedRose app = new GildedRose(items);

        int days = 2;
        if (args.length > 0) {
            days = Integer.parseInt(args[0]) + 1;
        }

        for (int i = 0; i < days; i++) {
            System.out.println("-------- day " + i + " --------");
            System.out.println("name, sellIn, quality");
            for (Item item : items) {
                System.out.println(item);
            }
            System.out.println();
            app.updateQuality();
        }
    }

}

File: python/texttest_fixture.py
# -*- coding: utf-8 -*-
from __future__ import print_function

from shi_mart import *

if __name__ == "__main__":
    print ("OMGHAI!")
    items = [
             Item(name="Bananas", sell_in=10, quality=20),
             Item(name="Aged Brie", sell_in=2, quality=0),
             Item(name="Eggs", sell_in=5, quality=7),
             Item(name="Eggs", sell_in=12, quality=5),
             Item(name="Canned Beans", sell_in=0, quality=80),
             Item(name="Canned Beans", sell_in=-1, quality=80),
        
             Item(name="Baked Sourdough Bread", sell_in=3, quality=6),  # <-- :O
            ]

    days = 30
    import sys
    if len(sys.argv) > 1:
        days = int(sys.argv[1]) + 1
    for day in range(days):
        print("-------- day %s --------" % day)
        print("name, sellIn, quality")
        for item in items:
            print(item)
        print("")
        ShiMart(items).update_quality()
